Sources/PackageCollections/Providers/GitHubPackageMetadataProvider.swift

```diff
@@ -167,23 +168,72 @@ struct GitHubPackageMetadataProvider: PackageMetadataProvider {
                         self.observabilityScope.emit(warning: "Failed to save GitHub metadata for package \(identity) to cache: \(error)")
                     }
 
-                    callback(.success(model))
+                    callback(.success(model), self.createContext(apiHost: baseURL.host, error: nil))
                 }
             } catch {
-                return callback(.failure(error))
+                self.errorCallback(error, apiHost: baseURL.host, callback: callback)
             }
         }
     }
-
-    func getAuthTokenType(for location: String) -> AuthTokenType? {
-        guard let baseURL = Self.apiURL(location) else {
+    
+    private func errorCallback(
+        _ error: Error,
+        apiHost: String?,
+        callback: @escaping (Result<Model.PackageBasicMetadata, Error>, PackageMetadataProviderContext?) -> Void
+    ) {
+        callback(.failure(error), self.createContext(apiHost: apiHost, error: error))
+    }
```

Comment:
would we need add something similar to this to the index client, or perhaps because it has no planned auth we dont really need to?

Reply:
> perhaps because it has no planned auth we dont really need to?

That's right. Package index client sets `PackageMetadataProviderContext` too, but since its errors are never authn related they don't need to be added to the context.